### ISSUE TITLE
Taint Analysis: `spl_object_id`-keyed removal bridges can strip taint off an unrelated node (false negatives)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1365]
Taint handlers that bridge `BeforeExpressionAnalysisInterface` to `RemoveTaintsInterface` record the argument node in `beforeExpressionAnalysis` and read it back in `removeTaints`, because `AddRemoveTaintsEvent` exposes neither a method id nor an argument offset. `WhereColumnTaintHandler` keys those records on `spl_object_id` and flushes them in `beforeAnalyzeFile`.

Its class docblock states the invariant that makes this safe:

> within one file the AST stays alive, so an id cannot be reused mid-file

That does not hold.

## Why

Psalm parses, analyses and frees whole foreign ASTs while the current file is still being analysed, and dispatches no `BeforeFileAnalysisEvent` for them:

- `ProjectAnalyzer::getMethodMutations()` builds an uncached `FileAnalyzer` for another file, analyses a method body, then releases it. Reached from `CallAnalyzer::collectSpecialInformation()` whenever a context collects mutations or initialisations, i.e. constructor property-initialisation checking on most classes.
- `ClassLikes::getTraitNode()` parses a whole file and keeps only its `Trait_` node.
- `StatementsProvider::getStatementsForFile()` caches no nodes, so each call returns a fresh object graph.

Those bodies do reach `beforeExpressionAnalysis`, and the gate is the global `$codebase->taint_flow_graph`, so a record is written there too. The entry then outlives its node, PHP reissues the freed handle, and an unrelated later node hits the stale record.

## Direction

A stale hit STRIPS taint, so the failure is a missed finding, not a false positive. For `WhereColumnTaintHandler` that is `sql`, i.e. a missed injection.

## Handle reuse is deterministic

```php
$recorded[spl_object_id($a)] = null;   // the StaticCall path retains no reference
unset($a);                             // foreign AST freed
$b = new PhpParser\Node\Expr\Variable("userSuppliedName");
// spl_object_id($b) === the recorded id
```

## Scope

- `src/Handlers/Eloquent/WhereColumnTaintHandler.php` (both `$whereColumnArgumentIds` and `$boundValuePositionIds`).
- The same pattern was introduced for `ResponseFactoryTaintHandler` in #1348 and is fixed on that branch.

## Fix

Key the maps weakly on the node object so an entry cannot outlive its key. `beforeAnalyzeFile` stays as a footprint cap that also survives a mid-file analysis throw, rather than as the collision guard.

Related: #734, #1300, #1306, PR #1218.
